telethon/client/telegrambaseclient.py

```diff
@@ -329,6 +320,9 @@ def __missing__(self, key):
         self.api_id = int(api_id)
         self.api_hash = api_hash
 
+        if inspect.iscoroutinefunction(self.session.process_entities):
+            self._log[__name__].warning('Using async sessions support is an experimental feature')
```

Comment:
I was more thinking of a https://docs.python.org/3/library/warnings.html in `utils.maybe_async`. So that it would only be printed once, whenever it first encountered an async method.